Include helper methods for the LanguagePlatform.

Impertatore opened the pull request:
The SegmentPair processer offers the possiblity to recover additional information from the ISegmentPair (FileTypeSupport.Framework)
through the LanguagePlatform, that includes the tokenized segments and word counts.

The processor creates a tempoaray translation memory given the settings provided; the TM is then used to recover the LanguagePlatform
segment, which is a sequence of Sdl.LanguagePlatform.Core.SegmentElements.

Impertatore committed:
Includs helper methods for the LanguagePlatform.

The SegmentPair processer offers the possiblity to recover additional information from the ISegmentPair (FileTypeSupport.Framework)
through the LanguagePlatform, that includes the tokenized segments and word counts.

The processor creates a tempoaray translation memory given the settings provided; the TM is then used to recover the LanguagePlatform
segment, which is a sequence of Sdl.LanguagePlatform.Core.SegmentElements.

Impertatore committed:
Minor refactoring: moved SegmentVisitor to Visitors directory

Impertatore committed:
Minor refactoring: Restructure the code environment; remove 'TranslationMemory' namespace

Impertatore committed:
Merge pull request #7 from Impertatore/master-LanguagePlatform.Helpers

Master language platform.helpers

Impertatore committed:
resolve error in replacing the parameters in the file name from the GetTemporaryTmPath function

Impertatore committed:
Merge pull request #8 from Impertatore/master-LanguagePlatform.Helpers

resolve error in replacing the parameters in the file name from the GetTemporaryTmPath function

Impertatore committed:
Minor refactoring; updated comments

Impertatore committed:
Merge pull request #9 from Impertatore/master-LanguagePlatform.Helpers

Minor refactoring; updated comments

Impertatore committed:
Updated the description for sdl.Community.Toolkit.LanguagePlatformin in the readme

Impertatore committed:
Merge pull request #10 from Impertatore/master-LanguagePlatform.Helpers

Updated the description for sdl.Community.Toolkit.LanguagePlatformin …

cromica commented on Sdl.Community.Toolkit.LanguagePlatform/Models/PathInfo.cs:
```diff
@@ -0,0 +1,64 @@
+				if (!string.IsNullOrEmpty(_productPath))
+					return _productPath;
+
+				_productPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), _productName);
```
Shouldn't we use the Studio folder path? The toolkit is not a product. The toolkit core has a bunch of libraries that provide the information required for identifying current version of Studio.

Impertatore replied:
The path information where the temporary resources are created and managed to provide this functionality is the responsibility of the client that is integrating with the functionality offered by the toolkit, i.e. thier product. The class called PathInfo includes default paths (and names), however this is the developers choice.

Wouldn’t be more correct to allow the developer to decide where these temporary resources are created & managed, with the expected defaults?

Impertatore replied:
Just thinking this through again and might make more sense to manage the temporary resources used in providing the functionality in the Studio folder path as you suggested.  I'll update the PR later on this evening

cromica replied:
Thanks Patrick. I agree that the developer should be able to decide about the path so having the PathInfo virtual is great. It's just around the default value.

Impertatore committed:
Resolved bugs in locating correct StudioVersion details; identified when multiple versions installed.
- Updated linq expression: where ExecutableVersion = Major (as opposed to MajorRevision)
- Removed: surplus directory seprators when loading the assembly file

Impertatore committed:
Merge pull request #11 from Impertatore/master-toolkit.core

Resolved bugs in locating correct StudioVersion details

Impertatore committed:
Recover the default product name from the version of Studio that is identified from the current app domain.
Note: Included a hard reference to the Sdl.Community.Toolkit.Core project/assembly, as I needed the update related to the PR that I submitted here https://github.com/sdl/SDL-Studio-Community-Toolkit/pull/4; might make sense to import the updated nuget package when available.

Impertatore committed:
Merge pull request #12 from Impertatore/master-LanguagePlatform.Helpers

Recover the default product name from the version of Studio
